src/opcodes/execution/far_call.rs

```diff
@@ -469,8 +469,10 @@ impl<const N: usize, E: VmEncodingMode<N>> DecodedOpcode<N, E> {
                 // we also do not return back cost of decommittment as it was subtracted
                 (MemoryPage(UNMAPPED_PAGE), remaining_ergs_of_caller_frame)
             } else {
-                assert!(normalized_preimage.0 != [0u8; 28], "original buffer {:?} lead to zero normalized preimage, but didn't trigger exception", buffer);
-
+                assert!(
```

Comment:
Imo we should do:

```rust
assert!(header != VersionedHashHeader::default(), "original buffer {:?} lead to zero header, but didn't trigger exception", buffer);
```

Otherwise we may have an empty header (guaranteed to be invalid) with a non-empty body.